pkg/endpoints/pipeline_test.go

```diff
@@ -1145,3 +1085,245 @@ func TestPipelineRunUpdateStatusAlreadySet412(t *testing.T) {
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "Pipeline1",
+		},
+		Spec: v1alpha1.PipelineSpec{},
+	}
+
+	_, err := r.PipelineClient.TektonV1alpha1().Pipelines("ns1").Create(&pipeline1)
+	if err != nil {
+		t.Errorf("FAIL: error creating test pipeline: %s", err)
+	}
+
+	httpWriter := httptest.NewRecorder()
+
+	pipelineRunBody := strings.NewReader(`{"pipelinename": "Pipeline1"}`)
+
+	pipelineRunRequest := dummyHTTPRequest("POST", "http://wwww.dummy.com:8383/v1/namespaces/ns1/pipelinerun", pipelineRunBody)
+	pipelineRunRequestRestful := dummyRestfulRequest(pipelineRunRequest, "ns1", "")
+	resp := dummyRestfulResponse(httpWriter)
+	r.createPipelineRun(pipelineRunRequestRestful, resp)
+
+	if resp.StatusCode() != 204 {
+		t.Errorf("FAIL: should have been recognised as a 204, got %d", resp.StatusCode())
+	}
+}
+
+/* PipelineRun no Pipeline 412 */
+func TestCreatePipelineRunNoPipeline(t *testing.T) {
+
+	r := dummyResource()
+
+	httpWriter := httptest.NewRecorder()
+
+	pipelineRunBody := strings.NewReader(`{"pipelinename": "Pipelinedoesnotexist"}`)
+
+	pipelineRunRequest := dummyHTTPRequest("POST", "http://wwww.dummy.com:8383/v1/namespaces/ns1/pipelinerun", pipelineRunBody)
+	pipelineRunRequestRestful := dummyRestfulRequest(pipelineRunRequest, "ns1", "")
+	resp := dummyRestfulResponse(httpWriter)
+	r.createPipelineRun(pipelineRunRequestRestful, resp)
+	if resp.StatusCode() != 412 {
+		t.Errorf("FAIL: should have been recognised as a 412, got %d", resp.StatusCode())
+	}
+}
+
+/* PipelineRun create with Git resource */
+func TestCreatePipelineRunGitResource(t *testing.T) {
+
+	r := dummyResource()
+
+	pipeline1 := v1alpha1.Pipeline{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "Pipeline1",
+		},
+		Spec: v1alpha1.PipelineSpec{},
+	}
+
+	_, err := r.PipelineClient.TektonV1alpha1().Pipelines("ns1").Create(&pipeline1)
+	if err != nil {
+		t.Errorf("FAIL: error creating test pipeline: %s", err)
+	}
+
+	httpWriter := httptest.NewRecorder()
+
+	pipelineRunBody := strings.NewReader(
+		`{"pipelinename":    "Pipeline1",
+			"gitresourcename": "git-source",
+			"gitcommit":       "12345",
+			"repourl":         "http://github.com/testorg/testrepo"}`)
+
+	pipelineRunRequest := dummyHTTPRequest("POST", "http://wwww.dummy.com:8383/v1/namespaces/ns1/pipelinerun", pipelineRunBody)
+	pipelineRunRequestRestful := dummyRestfulRequest(pipelineRunRequest, "ns1", "")
+	resp := dummyRestfulResponse(httpWriter)
+	r.createPipelineRun(pipelineRunRequestRestful, resp)
+
+	pipelineResourceList, err := r.PipelineClient.TektonV1alpha1().PipelineResources("ns1").List(metav1.ListOptions{})
+
+	numberOfGitResources := 0
+
+	for _, pipelineResource := range pipelineResourceList.Items {
+		if pipelineResource.Spec.Type == "git" {
+			numberOfGitResources++
+		}
+	}
+
+	if numberOfGitResources != 1 {
+		t.Errorf("FAIL: expected to find a single created PipelineResource of type Git, but found the number of Git PipelineResources to be %d", numberOfGitResources)
+	}
```

Comment:
Instead of just checking the number of Git resources, can you please verify that the PipelineResource created has the same content as what you provided (resource name prefix, gitcommit, repourl).